src/main/java/com/gitblit/wicket/pages/SessionPage.java

```diff
@@ -96,7 +96,9 @@ private void login() {
 					.getAttribute(Constants.AUTHENTICATION_TYPE);
 
 			// issue 62: fix session fixation vulnerability
-			session.replaceSession();
+			if(app().settings().getBoolean(Keys.web.rewriteSession, true)) {
```

Comment:
I won't merge this PR.  But I understand your issue and I propose the following alternative.
```java
if (AuthenticationType.CONTAINER != authenticationType) {
    session.replaceSession();
}
```

If that's acceptable, open a PR with that change and I'll merge it.